app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -59,7 +59,9 @@ def __init__(self, *args, **kwargs):
 
     def clean_json(self):
         json = self.cleaned_data["json"]
-        viewport_names = [x["viewport_name"] for x in json]
+        viewport_names = [
+            x["viewport_name"] for x in json if "viewport_name" in x

```

Comment:
`json` should not be used as a variable name as it shadows the built in module which is imported on L1. `x` is not a very clear variable name, `viewport` might be a better one but I am not sure what this "json" contains (that also could be better named). If a key only possibly exists in a dictionary you can use `foo.get("viewport_name")` which will return `None` by default. See https://realpython.com/python-lbyl-vs-eafp/. If an error found in production is being fixed a test should be included to ensure that it does not happen again.

Reply:
Done